metrics.go

```diff
@@ -276,9 +276,15 @@ func (mh *MetricsHandler) updateStats() {
 
 func (mh *MetricsHandler) startUpdatingStats() {
 	defer func() {
-		err := recover()
-		if err != nil {
-			mh.log.Fatalfln("Panic in metric updater: %v\n%s", err, string(debug.Stack()))
+		r := recover()
+		if r != nil {
+			evt := mh.log.Fatal()
+			if err, ok := r.(error); ok {
+				evt = evt.Stack().Err(err)
```

Comment:
I don't think `.Stack()` actually does anything, need to always use `debug.Stack()`

Reply:
Gah, you're right -- it only works with `pkg/errors`.